docs/data/data-grid/editing/editing.md

```diff
@@ -174,6 +174,24 @@ The value returned is used later as an argument on a call to `apiRef.current.upd
 />
 ```
 
+If you want to delete a row from the internal state of the Data Grid, you can return an additional property `_action: 'delete'` in the row object from the `processRowUpdate` callback. This will remove the row from the internal state of the Data Grid.
+It is a more performant way to delete a row as compared to updating the [`rows` prop](/x/react-data-grid/row-updates/#the-rows-prop) or using `setRows` API method because `processRowUpdate` uses the [`updateRows`](https://mui.com/x/react-data-grid/row-updates/#the-updaterows-method) under the hood which doesn't cause a full regeneration of the row tree.
+
+```tsx
+<DataGrid
+  // ...
```

Comment:
@MBilalShafi we could also do it like this to show that there are others, but are intentionally left out for readability

On second thought we could actually do write that there
```suggestion
  // remaining props intentionally left out for readability
```

WDYT?